File: app_canibuythat/src/main/java/com/gb/canibuythat/ui/ProgressDialog.java
package com.gb.canibuythat.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class ProgressDialog extends BaseDialogFragment {

    private static final String EXTRA_MESSAGE = "message";

    public static ProgressDialog newInstance(String message) {
        ProgressDialog progressDialog = new ProgressDialog();
        Bundle args = new Bundle();
        args.putString(EXTRA_MESSAGE, message);
        progressDialog.setArguments(args);
        return progressDialog;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = super.onCreateView(inflater, container, savedInstanceState);
        Bundle b = getArguments();
        setMessage(b.getString(EXTRA_MESSAGE));
        positiveBtn.setVisibility(View.GONE);
        activityIndicator.setVisibility(View.VISIBLE);
        return view;
    }
}

File: app_canibuythat/src/main/java/com/gb/canibuythat/ui/ProgressDialog.kt
package com.gb.canibuythat.ui

import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.prompt_dialog_layout.*

class ProgressDialog : BaseDialogFragment() {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        arguments?.getString(EXTRA_MESSAGE)?.let { setMessage(it) }
        button.visibility = View.GONE
        progress_bar.visibility = View.VISIBLE
    }

    companion object {
        const val EXTRA_MESSAGE = "message"

        fun newInstance(message: String) = ProgressDialog().apply {
            arguments = Bundle().apply {
                putString(EXTRA_MESSAGE, message)
            }
        }
    }
}
